### PR TITLE
[feat] Add target abstractions for release,package,sign

### PR DESCRIPTION
**Pull request by phoinixgrr**
#### Summary

We are adding abstractions for the following targets: 

```
make release                        to build and release artifacts
make package                        to build, package and push the artifact to a container registry
make sign                           to sign the artifact and perform verification
```

#### Ticket Link

Ticket: https://mattermost.atlassian.net/browse/DOPS-908

### CODE REVIEW

**Review by pfltdv — APPROVED** [review]
LGTM

**Comment by phoinixgrr** [comment]
@pfltdv  Could you please re-review . Pushed https://github.com/mattermost/rtcd/pull/55/commits/c9418c1cb1fd74635642a57d052b240e8e3576cf

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@ APP_NAME    := $(shell basename -s .git `git config --get remote.origin.url`)
 APP_COMMIT  := $(shell git log --pretty=format:'%h' -n 1)
 # Check if we are in protected branch, if yes use `protected_branch_name-sha` as app version.
 # Else check if we are in a release tag, if yes use the tag as app version, else use `dev-sha` as app version.
-APP_VERSION := $(shell if [ $(PROTECTED_BRANCH) = $(CURRENT_BRANCH) ]; then echo $(PROTECTED_BRANCH)-$(APP_COMMIT); else (git describe --abbrev=0 --exact-match --tags || echo dev-$(APP_COMMIT)) ; fi)
+APP_VERSION := $(shell if [ $(PROTECTED_BRANCH) = $(CURRENT_BRANCH) ]; then echo $(PROTECTED_BRANCH)-$(APP_COMMIT); else (git describe --abbrev=0 --exact-match --tags  > /dev/null 2>&1 || echo dev-$(APP_COMMIT)) ; fi)
```
Comment (pfltdv) [inline Makefile:18]: Sweet

**Review by pfltdv — APPROVED** [review]
LGTM

**Review by spirosoik — APPROVED** [review]
lgtm

**Pull request merged**